src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_AssignmentOperator.cs

```diff
@@ -260,7 +262,8 @@ private BoundExpression VisitAssignmentOperator(BoundAssignmentOperator node, bo
             ImmutableArray<int> argsToParamsOpt,
             BoundExpression rewrittenRight,
             TypeSymbol type,
-            bool used)
+            bool used,
+            Binder argumentsBinderOpt)
         {
```

Comment:
Should we add an assert that either this is not an index property or `argumentsBinderOpt` isn't `null`?